psi4/src/psi4/libmints/mintshelper.cc

```diff
@@ -785,11 +774,11 @@ SharedMatrix MintsHelper::so_dkh(int dkh_order) {
     return dkh;
 }
 
-SharedMatrix MintsHelper::ao_helper(const std::string &label, std::shared_ptr<TwoBodyAOInt> ints) {
-    std::shared_ptr<BasisSet> bs1 = ints->basis1();
-    std::shared_ptr<BasisSet> bs2 = ints->basis2();
-    std::shared_ptr<BasisSet> bs3 = ints->basis3();
-    std::shared_ptr<BasisSet> bs4 = ints->basis4();
+SharedMatrix MintsHelper::ao_helper(const std::string &label, std::vector<std::shared_ptr<TwoBodyAOInt>> ints) {
+    std::shared_ptr<BasisSet> bs1 = ints[0]->basis1();
+    std::shared_ptr<BasisSet> bs2 = ints[0]->basis2();
+    std::shared_ptr<BasisSet> bs3 = ints[0]->basis3();
+    std::shared_ptr<BasisSet> bs4 = ints[0]->basis4();
```

Comment:
Are you sure the basis sets are the same? Add a check and exception

Reply:
Because the vector is formed by creating clones of the first TwoBodyAOInt object, the basis sets should be the same.

```
std::vector<std::shared_ptr<TwoBodyAOInt>> ints_vec(nthread_);
    ints_vec[0] = std::shared_ptr<TwoBodyAOInt>(intf.f12_squared(exp_coeff));
    for (size_t i = 1; i < nthread_; i++) {
        ints_vec[i] = std::shared_ptr<TwoBodyAOInt>(ints_vec.front()->clone());
    }
```